File: NotesAppJv/app/src/main/java/com/henriquesbraga/notesappjv/api/ApiInterface.java
package com.henriquesbraga.notesappjv.api;

import com.henriquesbraga.notesappjv.model.Note;

import java.util.List;

import okhttp3.RequestBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface ApiInterface {
    @Headers("Content-Type: application/json")
    @POST("insert/")
    Call<Note> saveNote(@Body RequestBody params);

    @GET("get/all/")
    Call<List<Note>> getNotes();

    @Headers("Content-Type: application/json")
    @PUT("update/")
    Call<Note> updateNote(@Body RequestBody params);

    @Headers("Content-Type: application/json")
    @DELETE("delete/{id}")
    Call<Note> deleteNote(@Path("id") int id);
}

File: NotesAppKt/app/src/main/java/com/henriquesbraga/notesappkt/api/ApiInterface.kt
package com.henriquesbraga.notesappkt.api

import com.henriquesbraga.notesappkt.model.Note
import okhttp3.RequestBody
import retrofit2.Call
import retrofit2.http.*

interface ApiInterface {
    @Headers("Content-Type: application/json")
    @POST("insert/")
    fun saveNote(@Body params: RequestBody): Call<Note>

    @GET("get/all/")
    fun getNotes(): Call<ArrayList<Note>>

    @Headers("Content-Type: application/json")
    @PUT("update/")
    fun updateNote(@Body params: RequestBody): Call<Note>

    @Headers("Content-Type: application/json")
    @DELETE("delete/{id}")
    fun deleteNote(@Path("id") id: Int): Call<Note>
}